### PR TITLE
Prepare Geo Coded DataFrame memory reduction

### PR DESCRIPTION
**Pull request by gmwzest**
This modifies the Geo Coding prepare outputs to reduce the memory footprint.  Columns were being stored with string objects that require vastly more memory to store than the useful information they actually contain in bytes.  By converting to floating point and category types the memory footprint is reduced considerably.  This also adds early stopping and custom modeling parameters for the xgboost model builds.

### CODE REVIEW

File: zrp/modeling/pipeline_builder.py [hunk 1]
```diff
@@ -118,32 +148,87 @@ def __init__(self, zrp_model_source, file_path=None, zrp_model_name='zrp_0', *ar
...
-        self.zrp_model.fit(
-            X, y[self.race],
-            sample_weight=y.sample_weight
-        )
-
-        self.y_unique = y[self.race].unique()
+        print('\n---\nfitting zrp_model... n_class={}'.format(num_class))
+        #save_path = '/d/shared/users/gmw/zrp/model_artifacts'
+        #save_feather(X, save_path, "fe_data_{}.feather".format(self.zrp_model_source))
+        #save_feather(y[self.race], save_path, "target_data_{}.feather".format(self.zrp_model_source))
```
Comment (kasey-zest) [inline zrp/modeling/pipeline_builder.py:205]: Remove commented out saves

File: zrp/modeling/pipeline_builder.py [hunk 2]
```diff
@@ -190,9 +275,13 @@ class ZRP_DataSampling(BaseZRP):
         Indicates the source of zrp_modeling data to use. There are three optional sources 'block_group', 'census_tract', and 'zip_code'. By default 'census_tract' is inferred.
     population_weights_dict: dict
         Prevalence of target classes within the USA population as provided by the end-user. Sum of the values provided in the dictionary must be equal to one. Example: {'class1': 0.7, 'class2': 0.3}
+    test_size: float (default=0.2)
+        The fraction of samples to use as the test holdout
+    valid_size: float (default=0.2)
+        The fraction of samples to use as the test holdout
```
Comment (kasey-zest) [inline zrp/modeling/pipeline_builder.py:281]: `valid_size` update docstring

File: zrp/modeling/pipeline_builder.py [hunk 3]
```diff
@@ -254,14 +351,27 @@ class ZRP_Build(BaseZRP):
         Path indicating where to put artifacts folder its files (pipeline, model, and supporting data), generated during intermediate steps.
     zrp_model_name: str
         Name of zrp_model.
+    test_size: float (default=0.2)
+        The fraction of samples to use as the test holdout
+    valid_size: float (default=0.2)
+        The fraction of samples to use as the test holdout
```
Comment (kasey-zest) [inline zrp/modeling/pipeline_builder.py:357]: update docstring `valid_size`

File: zrp/modeling/pipeline_builder.py (hunk 2 above)
Comment (gmwzest) [reply zrp/modeling/pipeline_builder.py:278]: @kasey-zest by default though there is no validation set used... hence why the default is None

Comment (gmwzest) [reply zrp/modeling/pipeline_builder.py:278]: I switched it to default valid_size=0.0

**Review by kasey-zest — APPROVED** [review]
